src/avro_json_decoder.erl

```diff
@@ -212,7 +212,7 @@ parse_record_field(Attrs) ->
   Aliases   = avro_util:get_opt(<<"aliases">>, Attrs, []),
   FieldType = parse_schema(Type),
   #avro_record_field
-  { name    = Name
+  { name    = ?NAME(Name)
```

Comment:
Is this a fix for a pre-existing bug?

Reply:
the thing is that each Avro type has a constructor function, eg `avro_record:type/2`, `avro_array:type'2` etc. And each of those constructors does make sure that `Name` is a binary. But here in `avro_json_decoder` we don't use the constructors, but we build the record explicitly and not calling `?NAME` which makes sure it's `binary()`.
Also, the typespec for `#avro_record_field.name` says it should be `binary()`.

I'm quite surprised the `Name` was coming as a list here instead. I didn't try to trace back why was it string. Maybe it depends on JSON parser library settings

Reply:
And I had to enforce the conversion to binary because in this PR we started to compare `.name` and `.aliases` which obviously does not work if one of them is binary and another is string.